### ISSUE TITLE
[BUG][TGL][CI][SDW] IO transfer timed out in test check-kmod-load-unload-50.sh

### ISSUE THREAD

**Issue by aiChaoSONG** [thesofproject/linux#2552]
In daily report 666, `IO transfer timed out` is report on TGL laptop with SDCA codecs in test case check-kmod-load-unload-50.sh
```
sh-tgl-unb-sdw-01 kernel: [ 8410.417728] intel-sdw intel-sdw.0: IO transfer timed out, cmd 2 device 0 addr 50 len 6
sh-tgl-unb-sdw-01 kernel: [ 8410.417744] soundwire sdw-master-0: trf on Slave 0 failed:-110
sh-tgl-unb-sdw-01 kernel: [ 8410.417753] soundwire sdw-master-0: DEVID read fail:-110
sh-tgl-unb-sdw-01 kernel: [ 8410.417759] soundwire sdw-master-0: Slave attach failed: -110
```

**Recipe**
Kernel Branch: topic/sof-dev
Kernel Commit: 9055b6e3
SOF Branch: master
SOF Commit:  2aa62e1e
TPLG:  sof-tgl-rt711-rt1316-rt714.tplg
Platform: TGL-U laptop with SDCA codecs (2 ALC1316 amplifiers on separate links)

**Comment by naveen-manohar** [comment]
@aiChaoSONG For Chrome, we are in SoF_v010 1st drop and dint observe this failure.
Can you please help with the scenario being tried.. was this during a stress test ?

jfyi, we use sof-tgl-sdw-max98373-rt5682.tplg soundcard

**Comment by aiChaoSONG** [comment]
Observed in report 732 on Dell Notebook SDCA.

**Comment by plbossart** [comment]
@aiChaoSONG the error is completely different to the first one, this happens on a suspend not on an enumeration.

````
kernel: [ 8409.616187] sof-audio-pci 0000:00:1f.3: ipc tx succeeded: 0x40010000: GLB_PM_MSG: CTX_SAVE
kernel: [ 8409.616333] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x20140000 successful
kernel: [ 8411.671279] intel-sdw intel-sdw.0: IO transfer timed out, cmd 3 device 1 addr 5c len 1
kernel: [ 8411.671296] soundwire sdw-master-0: trf on Slave 1 failed:-110
kernel: [ 8413.718225] intel-sdw intel-sdw.0: IO transfer timed out, cmd 3 device 1 addr 5d len 1
kernel: [ 8413.718243] soundwire sdw-master-0: trf on Slave 1 failed:-110
kernel: [ 8413.719376] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x9010f0f successful
````

Can you please double-check why we don't have dynamic debug traces on this machine for the SoundWire code? We would want

````
options soundwire_bus dyndbg=+p
options soundwire_generic_allocation dyndbg=+p
options soundwire_cadence dyndbg=+p
options soundwire_intel_init dyndbg=+p
options soundwire_intel dyndbg=+p
````

Address 0x5c is SCP_SDCA_IntMask1, we'd need to check when this is set and with more traces figure out which codec driver tries to set/unset that bit.

**Comment by mengdonglin** [comment]
Very low reproduce rate. @RanderWang ran 10000+ iterations and cannot reproduce.

**Comment by RanderWang** [comment]
@plbossart Can we add a feature to do hardware reset to recover ? For massive production case,  it is inevitable to get such error.

**Comment by plbossart** [comment]
@aiChaoSong @RanderWang @mengdonglin, we see a timeout on the first iteration of check-kmod-load-unload-50.sh in Intel daily test #755. 

````
kernel: [ 8419.138667] intel-sdw intel-sdw.0: IO transfer timed out, cmd 3 device 1 addr 5c len 1
kernel: [ 8419.138683] soundwire sdw-master-0: trf on Slave 1 failed:-110
kernel: [ 8421.187746] intel-sdw intel-sdw.0: IO transfer timed out, cmd 3 device 1 addr 5d len 1
kernel: [ 8421.187762] soundwire sdw-master-0: trf on Slave 1 failed:-110
kernel: [ 8421.189059] sof-audio-pci 0000:00:1f.3: FW Poll Status: reg=0x9010f0f successful
````

I also cannot reproduce this issue so I am starting to wonder if this device has a problem or a different configuration?

**Comment by plbossart** [comment]
Since the timeouts only happen on the remove case, I added a possible fix in https://github.com/thesofproject/linux/pull/2563

@bardliao @RanderWang please review.

**Comment by mengdonglin** [comment]
@plbossart @bardliao @RanderWang This issue cannot be reproduced in latest daily test report ID 789. Let's monitor for a few more days to see if we can close this issue.

**Comment by aiChaoSONG** [comment]
Not reproduced on: 2020/11/23 UTC+8

**Comment by mengdonglin** [comment]
Not reproduced in daily test on 2020-11-24 UTC+8. Closed

**Comment by mengdonglin** [comment]
Reopen as this is can be covered by https://github.com/thesofproject/linux/issues/2578. Will revisit this issue after #2578 is fixed.

**Comment by aiChaoSONG** [comment]
reproduced in daily report 962(12/4/2020 utc+8) https://sof-ci.sh.intel.com/#/result/planresultdetail/962?model=TGLU_ANDR_SDCA&testcase=check-kmod-load-unload-50

**Comment by mengdonglin** [comment]
@aiChaoSONG @plbossart can we close this issue? PR #2563 was merged.

**Comment by plbossart** [comment]
closing, the timeouts reported in https://github.com/thesofproject/linux/issues/2552#issuecomment-738515455 are a separate issue tracked in https://github.com/thesofproject/linux/issues/2627
